### PR TITLE
Add infrastructure for unit tests

### PR DESCRIPTION
**Pull request by pudelkoM**
This PR adds the necessary infrastructure for Go unit tests.

### CODE REVIEW

**Comment by pudelkoM** [comment]
@osinstom

**Comment by krsna1729** [comment]
i think we can move it out of Dockerfile? just directly run `CGO_ENABLED=0 go test -mod=vendor -v pfcpiface` on the host instead of going via docker build?

**Comment by pudelkoM** [comment]
The CI base image (`runs-on: ubuntu-latest` ) does not have go installed, right?
So I put it inside the same container that we build and publish for reproducibility.

**Comment by krsna1729** [comment]
@pudelkoM https://github.com/omec-project/upf-epc/blob/master/.github/workflows/pull_request.yml#L24-L28

I feel like there is go installed?

**Comment by pudelkoM** [comment]
Right go should be there. I got that confused with the local development setup. There go might not be installed or outdated. 
To make testing as easy as building, it could run in docker.

**Comment by krsna1729** [comment]
Ok.

File: Makefile [hunk 1]
```diff
@@ -69,4 +69,9 @@ pb:
 		.;
 	cp -a output/bess_pb ${BESS_PB_DIR}
 
-.PHONY: docker-build docker-push output pb
+test:
+	DOCKER_BUILDKIT=$(DOCKER_BUILDKIT) docker build $(DOCKER_PULL) $(DOCKER_BUILD_ARGS) \
```
Comment (osinstom) [inline Makefile:73]: I think it's counterintuitive to use Docker build for running tests. + we produce an unnecessary Docker image. I propose to use `golang:1.13` image and use `docker run`. The [Antrea](https://github.com/antrea-io/antrea/blob/main/Makefile#L146) project is a good example.

Comment (krsna1729) [reply Makefile:73]: makes sense

**Pull request closed without merging**